libs/mocks/src/permissions.rs

```diff
@@ -0,0 +1,60 @@
+pub use pallet_mock_permissions::*;
+
+#[allow(dead_code)]
+#[frame_support::pallet]
+pub mod pallet_mock_permissions {
+	use cfg_traits::Permissions;
+	use cfg_types::permissions::Role;
+	use frame_support::pallet_prelude::*;
+	use mock_builder::{execute_call, register_call};
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Scope;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::storage]
+	pub(super) type CallIds<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		<Blake2_128 as frame_support::StorageHasher>::Output,
+		mock_builder::CallId,
+	>;
+
+	impl<T: Config> Pallet<T> {
+		pub fn mock_has(f: impl Fn(T::Scope, T::AccountId, Role) -> bool + 'static) {
+			register_call!(move |(a, b, c)| f(a, b, c));
+		}
+
+		pub fn mock_add(f: impl Fn(T::Scope, T::AccountId, Role) -> bool + 'static) {
+			register_call!(move |(a, b, c)| f(a, b, c));
+		}
+
+		pub fn mock_remove(f: impl Fn(T::Scope, T::AccountId, Role) -> bool + 'static) {
+			register_call!(move |(a, b, c)| f(a, b, c));
+		}
+	}
```

Comment:
Could also imagine something like
```suggestion
       #[mock_builder::builder]
	impl<T: Config> Pallet<T> {
	        #[builder::register]
		pub fn mock_has(f: impl Fn(T::Scope, T::AccountId, Role) -> bool;

		pub fn mock_add(f: impl Fn(T::Scope, T::AccountId, Role) -> bool + 'static) {
			register_call!(move |(a, b, c)| f(a, b, c));
		}

		pub fn mock_remove(f: impl Fn(T::Scope, T::AccountId, Role) -> bool + 'static) {
			register_call!(move |(a, b, c)| f(a, b, c));
		}
	}
```